tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/navigator/workspace/LiferayWorkspaceServerActionProvider.java

```diff
@@ -80,9 +90,108 @@ private Separator _invisibleSeparator(String s) {
 		return separator;
 	}
 
-	private void _makeActions(ISelectionProvider provider) {
+	private void _makeActions(IMenuManager menu, ISelectionProvider provider) {
+		ISelection selection = provider.getSelection();
+
+		if (selection == null) {
+			return;
+		}
+
 		_watchAction = new WatchWorkspaceModulesAction(provider);
 		_stopAction = new StopWorkspaceModulesAction(provider);
+
+		if (selection instanceof TreeSelection) {
+			TreeSelection watchSelection = (TreeSelection)selection;
+
+			List<?> selectProjects = watchSelection.toList();
+
+			if (ListUtil.hasMultipleValues(selectProjects)) {
+				_watchAction.setEnabled(true);
+				_stopAction.setEnabled(true);
+
+				menu.add(_watchAction);
+				menu.add(_stopAction);
+
+				return;
+			}
+
+			Object selectedProject = watchSelection.getFirstElement();
+
+			IWorkspaceProject workspaceProject = LiferayCore.create(IWorkspaceProject.class, selectedProject);
+
+			if (workspaceProject != null) {
+				IProject project = workspaceProject.getProject();
+
+				Set<IProject> watchingProject = workspaceProject.watching();
+
+				String jobName = project.getName() + ":watch";
```

Comment:
We need a better way to lookup watch jobs.  @jtydhr88 @Camork we need to do something with qualified names like we did in other tickets.  Can you show @jtydhr88 that technique?

Reply:
We can set a specific job family name to the watch tasks in [L145](https://github.com/gamerson/liferay-ide/blob/master/tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleWorkspaceProject.java#L145)